enclave/core/sgx/calls.c

```diff
@@ -505,6 +506,100 @@ oe_result_t oe_get_enclave_status()
     return __oe_enclave_status;
 }
 
+/*
+**==============================================================================
+**
+** _debug_write_exit_frame()
+**
+** If a debug enclave, write the exit frame information to host's
+** ecall_context so that the host can stitch the ocall stack.
+**
+** This function is intended to be called by oe_asm_exit (see below).
+** When called, the call stack would look like this:
+**
+**     enclave-function -> oe_ocall -> oe_asm_exit -> _debug_write_exit_frame
+**                                    __morestack
+**
+** The _debug_write_exit_frame reads its caller (oe_asm_exit/__morestack)
+** infomation (return address, rbp) and passes it along to the host in the
+** ecall_context.
+**
+** ocall invokes eexit instruction which resumes execution in host at the
+** oe_enter function. The host dispatches the ocall via the following sequence:
+**
+**     oe_enter -> __oe_host_stack_bridge -> __oe_dispatch_ocall
+**                                                    -> invoke ocall function
+**
+** Now that the enclave exit is available to the host, __oe_host_stack_bridge
+** temporarily modifies its caller info with the enclave's exit information so
+** that the stitched stack looks like this:
+**
+**      in enclave         |   in host
+** oe_ocall -> oe_asm_exit -> __oe_host_stack_bride -> __oe_dispatch_ocall ...
```

Comment:
A bit of cleanup:
 - `oe_asm_exit` should be `oe_exit_enclave`
 - `__oe_host_stack_bridge` has a typo

```
oe_ocall                                       | 
  -> oe_enclave_exit (aliased as __morestack)  | inside enclave
-------------------------------------------------------------------------------
    -> __oe_host_stack_bridge                  | outside enclave (host context)
      -> __oe_dispatch_ocall                   | 
        -> ...                                 |

Reply:
Haha. I wonder who "host stack bride" is :D

Reply:
Fixed. Great suggestion.